src/app/products/product.effects.ts

```diff
@@ -42,18 +44,30 @@ export class ProductEffects {
       ofType(PRODUCT_ACTIONS.loadSelectedProduct),
       switchMap(({ productId }) =>
         this.productService.getSelectedProduct(productId).pipe(
-          tap((product) => {
-            this.router.navigate(['/details', product.id]);
+          map((product) => {
+            return PRODUCT_ACTIONS.loadSelectedProductSuccess({ product });
           }),
-          map((product) => 
-            PRODUCT_ACTIONS.loadSelectedProductSuccess({ product })
-          ),
           catchError((error) =>
             of(PRODUCT_ACTIONS.loadSelectedProductFailure({ error }))
           )
         )
       )
     )
   );
-  
+
+  public searchProducts$ = createEffect(() =>
```

Comment:
this is not right here. 
The code is complicated. And again why effect?
Your service should handle all the logic such that in here you only subscribe.